gossip-base/src/main/java/org/apache/gossip/event/data/UpdateNodeDataEventHandler.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gossip.event.data;
+
+import java.util.List;
+
+/**
+ * Event handler interface for the per node data items.
+ * Classes which implement this interface get notifications when per node data item get changed
+ * based on the key list that class defined.
+ */
+public interface UpdateNodeDataEventHandler {
+  
+  /**
+   * This method get called when a per node datum get changed.
+   *
+   * @param nodeId   id of the node that change the value
+   * @param key      key of the datum
+   * @param oldValue previous value of the datum or null if the datum is discovered
+   *                 for the first time
+   * @param newValue updated value of the datum
+   */
+  void onUpdate(String nodeId, String key, Object oldValue, Object newValue);
+  
+  /**
+   * This method get called to retrieve the key list that implemented class should interested in.
+   * The implemented class only get notifications for the key list returned by this method.
+   *
+   * @return interested list of per node data keys that the implementation class will like
+   * to get notified
+   */
+  List<String> getNodeDataListeningKeys();
```

Comment:
The second method is for defining the interesting  key list for example some implemented class will define key a,b,c,d as interested  and that class will only get notified if the defined keys got changed.
However I also think the second approach is much nicer